### PR TITLE
WP-710: For apps with no inputs, only show launch button

### PR DESCRIPTION
**Pull request by chandra-tacc**
## Overview: ##
For certain apps like OpenSees-Interactive, the only input is maxMinutes. Provide a simple job launch option for such apps.
See design in [WP-710](https://tacc-main.atlassian.net/browse/WP-710) 

## PR Status: ##

* [X] Ready.
* [ ] Work in Progress.
* [ ] Hold.

## Related Jira tickets: ##

* [WP-710](https://tacc-main.atlassian.net/browse/WP-710)

## Summary of Changes: ##
1. Add App form notes option to hide max minutes. See [Apps PR 58](https://github.com/TACC/WMA-Tapis-Templates/pull/58). 
2. In DesignSafe Portal code, add notes optional fields for hideMaxMinutes and an launch text.
3. Hide Max Minutes when the notes flag indicates so.
4. When the steps have no fields (or we have no steps), provide a quick launch UI. The launch button varies between interactive and other apps.
5. jobLaunchDescription is unused in app, but could be useful if users/PI wants to add custom text.

## Testing Steps: ##
1. Open https://designsafe.dev/rw/workspace/opensees-interactive?appVersion=3.7.0 and see if you can launch the app with the launch button session. The UI matches what is shown in [WP-710](https://tacc-main.atlassian.net/browse/WP-710) 

2. Regression testing: https://designsafe.dev/rw/workspace/openfoam. In this app, the regular form should show. Also, in the configuration step, the field dependency works as normal (change from small to large to queues, varies maxMinutes and other fields).

3. Verified the json we submit from server logs:
```
op=submitJob info={'body': {'operation': 'submitJob', 'licenseType': None, 'isInteractive': True, 'job': {'archiveOnAppError': True, 'appId': 'opensees-interactive', 'appVersion': '3.7.0', 'execSystemId': 'wma-exec-01', 'fileInputs': [], 'parameterSet': {'envVariables': [{'key': '_UserProjects', 'value': ''}]}, 'name': 'opensees-interactive-3.7.0_2024-10-08T19:22:21', 'archiveSystemId': 'designsafe.storage.default', 'archiveSystemDir': 'foouser/tapis-jobs-archive/${JobCreateDate}/${JobName}-${JobUUID}'}}}
```

## UI Photos:
Screenshot 2024-10-08 at 2 21 51 PM

### CODE REVIEW

File: client/modules/workspace/src/AppsSubmissionForm/AppsSubmissionForm.tsx [hunk 1]
```diff
@@ -615,23 +620,47 @@ export const AppsSubmissionForm: React.FC = () => {
...
+                    <Space direction="vertical" size="large">
+                      <div>
+                        {definition.notes.jobLaunchDescription ??
+                          'This job is pre-configured. No input is necessary to submit the job.'}
+                      </div>
+                      <div>
+                        <PrimaryButton
+                          htmlType="submit"
+                          loading={isPending}
+                          style={{ width: 130 }}
```
Comment (rstijerina) [inline client/modules/workspace/src/AppsSubmissionForm/AppsSubmissionForm.tsx:634]:
```suggestion
                          style={{ width: 150 }}
```

Comment (rstijerina) [reply client/modules/workspace/src/AppsSubmissionForm/AppsSubmissionForm.tsx:634]: Could also make the submit button a separate component, and import it here from AppsSubmissionDetails

Comment (chandra-tacc) [reply client/modules/workspace/src/AppsSubmissionForm/AppsSubmissionForm.tsx:634]: good idea, will do

File: client/modules/workspace/src/AppsSubmissionForm/AppsSubmissionForm.tsx [hunk 2]
```diff
@@ -615,23 +620,47 @@ export const AppsSubmissionForm: React.FC = () => {
...
-                    setCurrent={setCurrent}
-                    definition={definition}
-                  />
-                </Col>
+                {Object.keys(steps || {}).length === 0 ? (
+                  <Col style={{ marginTop: '32px', marginLeft: '32px' }}>
+                    <Space direction="vertical" size="large">
+                      <div>
+                        {definition.notes.jobLaunchDescription ??
+                          'This job is pre-configured. No input is necessary to submit the job.'}
```
Comment (rstijerina) [inline client/modules/workspace/src/AppsSubmissionForm/AppsSubmissionForm.tsx:628]: @wesleyboar I wonder, should we have any post-launch text change or state change? i.e. make the button disabled and add a check mark indicating the session was launched. Might be good to note if the modal is closed.

File: client/modules/workspace/src/AppsSubmissionForm/AppsSubmissionForm.tsx (hunk 1 above)
Comment (chandra-tacc) [reply client/modules/workspace/src/AppsSubmissionForm/AppsSubmissionForm.tsx]: Added custom component and tested with 
* qgis desktop (launch button)
* Opensees interactive VM (simplified launch)
* OpenSees SP (standard job submit)

**Review by rstijerina — APPROVED** [review]

File: client/modules/workspace/src/AppsSubmissionForm/AppsSubmissionForm.tsx (hunk 2 above)
Comment (rstijerina) [reply client/modules/workspace/src/AppsSubmissionForm/AppsSubmissionForm.tsx:628]: @chandra-tacc Maybe we can use the same pattern here: https://github.com/DesignSafe-CI/portal/blob/e409cbe9bc0b2b7c033d08b22e681f09b78e9dbd/client/modules/workspace/src/JobsListing/JobsListing.tsx#L51

Comment (chandra-tacc) [reply client/modules/workspace/src/AppsSubmissionForm/AppsSubmissionForm.tsx:628]: Great idea @rstijerina. Made that change. Below are screenshots that show in various states.

### Launched with modal open. Launch is disabled.
Screenshot 2024-10-10 at 3 04 51 PM


### Launch with modal closed. 
Screenshot 2024-10-10 at 3 05 02 PM

### Launch after page refresh, resets to enabled.
Screenshot 2024-10-10 at 3 07 52 PM

Comment (rstijerina) [reply client/modules/workspace/src/AppsSubmissionForm/AppsSubmissionForm.tsx:628]: Looks great! Maybe also include the check mark?
https://github.com/DesignSafe-CI/portal/blob/e409cbe9bc0b2b7c033d08b22e681f09b78e9dbd/client/modules/workspace/src/JobsListing/JobsListing.tsx#L56

Comment (chandra-tacc) [reply client/modules/workspace/src/AppsSubmissionForm/AppsSubmissionForm.tsx:628]: There is more work, will need to make a bit more adjustment based on success state.

Comment (rstijerina) [reply client/modules/workspace/src/AppsSubmissionForm/AppsSubmissionForm.tsx:628]: @wesleyboar Should we also change the text once disabled, to:
"Session has been launched. You can view status in <a>Job Status</a>"

**Review by wesleyboar — CHANGES_REQUESTED** [review]
✓ This works and looks like design.
<details><summary>ⓧ Allows excess submissions.</summary>

excess submissions

https://github.com/user-attachments/assets/25a854f5-fad4-444d-af01-96d0946e84b4

Yes, I know it requires a special kind of user (too-hasty-to-read, impatient) to do this, but we've all likely seen people at a walk sign or an elevator press the button more than once. if we can easily stop them from draining resources, lets.

</details>

Please [change button behavior as @rstijerina suggested](https://github.com/DesignSafe-CI/portal/pull/1469#discussion_r1795871278). If user leaves this workspace form or page or website, returns to form, then button is again enabled.

File: client/modules/workspace/src/AppsSubmissionForm/AppsSubmissionForm.tsx (hunk 2 above)
Comment (wesleyboar) [reply client/modules/workspace/src/AppsSubmissionForm/AppsSubmissionForm.tsx:628]: Yeah, if we have the state, lets. Great idea.

**Comment by chandra-tacc** [comment]
> ✓ This works and looks like design.
> 
> ⓧ Allows excess submissions.
Thanks for reviewing, will handle excess submissions.

File: client/modules/workspace/src/AppsSubmissionForm/AppsSubmissionForm.tsx (hunk 2 above)
Comment (chandra-tacc) [reply client/modules/workspace/src/AppsSubmissionForm/AppsSubmissionForm.tsx]: @rstijerina  - done made the adjustments.
1. Button text has the check mark on job success.
2. Job success text changes based on state.

Screen recording of this:

https://github.com/user-attachments/assets/5ac47ccb-d412-4430-814a-96105ee137b7

**Review by wesleyboar — APPROVED** [review]
⭐️

https://github.com/user-attachments/assets/b138dc03-2f74-40e6-8142-9abeeae5feb2

**Pull request merged**